[googletts] Refactored to use openHAB Core OAuth2 (#6489)
openhab/openhab-addons: bundles/org.openhab.voice.googletts/README.md

# Google Cloud Text-to-Speech

> Google Cloud TTS Service uses the none-free Google Cloud Text-to-Speech API to convert text or Speech Synthesis Markup Language (SSML) input into audio data of natural human speech. 

Reviewer: ```suggestion
Google Cloud TTS Service uses the non-free Google Cloud Text-to-Speech API to convert text or Speech Synthesis Markup Language (SSML) input into audio data of natural human speech. 
```

> It provides 30 voices, available in multiple languages and variants and applies DeepMind’s groundbreaking research in WaveNet and Google’s powerful neural networks. 

Reviewer: They have added many more voices. According to https://cloud.google.com/text-to-speech:

> Google Cloud Text-to-Speech offers a selection of 180+ voices across 30+ languages and variants, enabling developers to pick the voice that works best for their application.

Author: I used a more general wording now. Without calling on numbers.

# Google Cloud Text-to-Speech

Google Cloud TTS Service uses the non-free Google Cloud Text-to-Speech API to convert text or Speech Synthesis Markup Language (SSML) input into audio data of natural human speech. 

> It provides multiple voices, available in diferent languages and variants and applies DeepMind’s groundbreaking research in WaveNet and Google’s powerful neural networks. 

Reviewer: ```suggestion
It provides multiple voices, available in different languages and variants and applies DeepMind’s groundbreaking research in WaveNet and Google’s powerful neural networks. 
```

Author: :roll_eyes: Good to have four eyes on it ... Thanks.

* Make sure that billing is enabled for your project. [link](https://cloud.google.com/billing/docs/how-to/modify-project)
* Enable the Cloud Text-to-Speech API. [link](https://console.cloud.google.com/apis/dashboard)
* Set up authentication:
  * Go to the "APIs & Services" -> "Credentials" page in the GCP Console and your project. [link](https://console.cloud.google.com/apis/credentials)
  * From the "Create credentials" drop-down list, select "OAuth client ID.
  * Select application type "Other" and enter a name into the "Name" field.
  * Click Create. A pop-up appears, showing your "client ID" and "client secret".

## Service Configuration

Using your favorite configuration UI (e.g. Paper UI) edit **Services/Voice/Google Cloud Text-to-Speech** settings and set 
* **Client Id** - Google Cloud Platform OAuth 2.0-Client Id.
* **Client Secret** - Google Cloud Platform OAuth 2.0-Client Secret.

> * **Authorization Code** - The auth-code is a one-time code needed to retrieve the necessary access-codes from Google Cloud Platform. <b>Please go to your broser ...</b> (https://accounts.google.com/o/oauth2/auth?client_id={{clientId}}&redirect_uri=urn:ietf:wg:oauth:2.0:oob&scope=https://www.googleapis.com/auth/cloud-platform&response_type=code)[https://accounts.google.com/o/oauth2/auth?client_id={{clientId}}&redirect_uri=urn:ietf:wg:oauth:2.0:oob&scope=https://www.googleapis.com/auth/cloud-platform&response_type=code] (replace {{clientId}} by your Client Id) <b>... to generate an auth-code and paste it here</b>. After initial authorization, this code is not needed anymore. It is recommended to clear this configuration parameter afterwards.

Author: ```suggestion
* **Authorization Code** - The auth-code is a one-time code needed to retrieve the necessary access-codes from Google Cloud Platform. **Please go to your browser ...** (https://accounts.google.com/o/oauth2/auth?client_id={{clientId}}&redirect_uri=urn:ietf:wg:oauth:2.0:oob&scope=https://www.googleapis.com/auth/cloud-platform&response_type=code)[https://accounts.google.com/o/oauth2/auth?client_id={{clientId}}&redirect_uri=urn:ietf:wg:oauth:2.0:oob&scope=https://www.googleapis.com/auth/cloud-platform&response_type=code] (replace `{{clientId}}` by your Client Id) **... to generate an auth-code and paste it here**. After initial authorization, this code is not needed anymore. It is recommended to clear this configuration parameter afterwards.
```
